runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -235,32 +261,60 @@ public int appendObjectState(Stream<Map.Entry> entryStream) {
         ImmutableMap<CheckpointEntry.CheckpointDictKey, String> mdkv =
                 ImmutableMap.copyOf(this.mdkv);
 
-        final Iterable<List<Map.Entry>> partitions =
-                Iterables.partition(entryStream::iterator, batchSize);
-        int entryCount = 0;
-
-        for (List<Map.Entry> partition : partitions) {
-            MultiSMREntry smrEntries = new MultiSMREntry();
-            for (Map.Entry entry : partition) {
-                smrEntries.addTo(new SMREntry("put",
-                        new Object[]{keyMutator.apply(entry.getKey()),
-                                valueMutator.apply(entry.getValue())},
-                        serializer));
-                entryCount++;
+        int totalEntryCount = 0;
+        int currentBatchSize = 0;
+        int numBytesPerCheckpointEntry = 0;
+        int maxWriteSizeLimit = (int) (BATCH_THRESHOLD_PERCENTAGE * rt.getParameters().getMaxWriteSize());
+
+        ByteBuf inputBuffer = Unpooled.buffer();
+        MultiSMREntry smrEntries = new MultiSMREntry();
+
+        for (Map.Entry entry : (Iterable<Map.Entry>) entryStream::iterator) {
```

Comment:
Casting to `Iterable` just to use an enhanced-for loop seems unnecessary here.

Reply:
@Maithem Do you have a better suggestion here? I got rid of one Iterable.partitions call from the original change and this is something I came up with that seemed a better version and suitably aligned to this change. Can you help me understand what might be better here with an example?

Reply:
Changed to use an iterator.